### PR TITLE
vmware virtual disk

### PR DESCRIPTION
**Pull request by maxmanuylov**
Done:
* Create/destroy disk
* Attach/detach disk to/from machine

Not done:
* Create disk in non-existing folder
* Disk refresh
* VM disks refresh
* Assign attached disk to the specific device (e.g. "/dev/sdb"; not sure it is possible though)

### CODE REVIEW

File: resource_virtual_machine.go [hunk 1]
```diff
@@ -228,6 +233,53 @@ func resourceVirtualMachineCreate(d *schema.ResourceData, meta interface{}) erro
...
+			diskDatastoreName := disk["datastore"].(string)
+
+			datastore, err := finder.Datastore(ctx, fmt.Sprintf("/%v/datastore/%v", dc_name, diskDatastoreName))
+			if err != nil {
+				return fmt.Errorf("Failed to find datastore \"%s\": %v", diskDatastoreName, err)
+			}
+
+			datastoreRef := datastore.Reference()
+
+			controllerDevice, err := devices.CreateSCSIController("")
```
Comment (VladRassokhin) [inline resource_virtual_machine.go:249]: Controller would be created to each disk?

**Review by VladRassokhin — COMMENTED** [review]
Seems good, yep please run `make fmt` :)

File: resource_virtual_machine.go (hunk 1 above)
Comment (maxmanuylov) [reply resource_virtual_machine.go:249]: Yes so far. Is it possible to have one for all disks?

Comment (VladRassokhin) [reply resource_virtual_machine.go:249]: AFAIK yes, as it's usual IDE/SATA/SCSI controller and supports many disks/cdroms/etc.
@mkuzmin don't you remember where's groovy code we created >year ago to copy/attach disks in?

**Pull request merged**